src/query/src/datafusion.rs

```diff
@@ -390,6 +395,54 @@ impl QueryExecutor for DatafusionQueryEngine {
     }
 }
 
+/// Creates a table in memory and executes a show statement on the table.
+pub async fn execute_show_with_filter(
+    record_batch: RecordBatch,
+    filter: Option<Expr>,
+    table_name: String,
+) -> Result<Output> {
+    let sql = if let Some(filter) = filter {
+        format!("SELECT * FROM {table_name} WHERE {}", filter)
+    } else {
+        format!("SELECT * FROM {table_name}")
+    };
```

Comment:
I'd rather use the dataframe [APIs](https://docs.rs/datafusion/latest/datafusion/dataframe/struct.DataFrame.html#method.filter) instead of `format!`-ing SQL.

Reply:
I can't address this review since `sqlparser::ast::Expr` is different from `datafusion_expr::expr::Expr`.  Do you know how to convert `sqlparser::ast::Expr` into `datafusion_expr::expr::Expr`? Or can I leave TODO this line?

Reply:
The query engine already has an API to read table by dataframe:

https://github.com/GreptimeTeam/greptimedb/blob/559d1f73a2819b775b9c8757cff08522ec216f14/src/query/src/query_engine.rs#L81

You can use it directly. Refer to the example that using it for prometheus query:

https://github.com/GreptimeTeam/greptimedb/blob/559d1f73a2819b775b9c8757cff08522ec216f14/src/servers/src/prometheus.rs#L65

Reply:
I've roughly implemented on the reviews. If you agree with this approach, I'll continue with the implementation.

Reply:
https://github.com/GreptimeTeam/greptimedb/pull/1829/files#diff-0118b58a418d785d52d3184c96f07bd4075e5c7e3966c506ec1ae482e595a289R383-R402

Reply:
@killme2008 

>The query engine already has an API to read table by dataframe:

We can't use it because tables which are referenced on the show statement get created in memory. So the tables don't implement `Table` trait.